src/Microsoft.AspNetCore.Routing/Matchers/DefaultEndpointSelector.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class DefaultEndpointSelector : EndpointSelector
```

Comment:
So my plan is that this is the `EndpointSelector` that we *want* to use. MVC will override it when it needs to.